pallets/runtime/tests/src/asset_metadata_test.rs

```diff
@@ -0,0 +1,427 @@
+            global_key,
+            value.clone(),
+            details.clone()
+        ));
+
+        // Set metadata value for local key.
+        exec_ok!(Asset::set_asset_metadata(
+            owner.origin(),
+            ticker,
+            local_key,
+            value.clone(),
+            details.clone()
+        ));
+
+        let value2 = AssetMetadataValue("beef".as_bytes().into());
+        let details2 = Some(make_metadata_value_details(Some(1), false));
+
+        // Try updating global metadata value with user that doesn't have permissions for this asset.
+        exec_noop!(
+            Asset::set_asset_metadata(
+                other.origin(),
+                ticker,
+                global_key,
+                value2.clone(),
+                details2.clone()
+            ),
+            EAError::UnauthorizedAgent
+        );
+
+        // Try updating local metadata value with user that doesn't have permissions for this asset.
+        exec_noop!(
+            Asset::set_asset_metadata(
+                other.origin(),
+                ticker,
+                local_key,
+                value2.clone(),
+                details2.clone()
+            ),
+            EAError::UnauthorizedAgent
+        );
+
+        // Update metadata value for global key.
+        exec_ok!(Asset::set_asset_metadata(
+            owner.origin(),
+            ticker,
+            global_key,
+            value2.clone(),
+            details2.clone(),
+        ));
+
+        // Update metadata value for local key.
+        exec_ok!(Asset::set_asset_metadata(
+            owner.origin(),
+            ticker,
+            local_key,
+            value2.clone(),
+            details2.clone(),
+        ));
+
+        let details_locked = make_metadata_value_details(None, true);
+        // Try locking global metadata value with user that doesn't have permissions for this asset.
+        exec_noop!(
+            Asset::set_asset_metadata_details(
+                other.origin(),
+                ticker,
+                global_key,
+                details_locked.clone()
+            ),
+            EAError::UnauthorizedAgent
+        );
+
+        // Try locking local metadata value with user that doesn't have permissions for this asset.
+        exec_noop!(
+            Asset::set_asset_metadata_details(
+                other.origin(),
+                ticker,
+                local_key,
+                details_locked.clone()
+            ),
+            EAError::UnauthorizedAgent
+        );
+
+        // Lock metadata value for global key.
+        exec_ok!(Asset::set_asset_metadata_details(
+            owner.origin(),
+            ticker,
+            global_key,
+            details_locked.clone(),
+        ));
+
+        // Lock metadata value for local key.
+        exec_ok!(Asset::set_asset_metadata_details(
+            owner.origin(),
+            ticker,
+            local_key,
+            details_locked.clone(),
```

Comment:
Potential for deduping with closure here too

Reply:
no.